File: Examples/test-suite/ruby/li_cdata_cpp_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#

require 'swig_assert'
require 'li_cdata_cpp'

include Li_cdata_cpp


s = "ABC\0abc"
m = malloc(256)
memmove(m, s)
ss = cdata(m, 7)
swig_assert(ss == "ABC\0abc", binding, "failed");

File: Examples/test-suite/csharp/li_cdata_cpp_runme.cs
using System;
using System.Text;
using li_cdata_cppNamespace;

public class li_cdata_cpp_runme {

  public static void Main() {
    Encoding utf8 = Encoding.UTF8;
    string s0 = "ABC\0abc";
    byte[] s = utf8.GetBytes(s0);
    SWIGTYPE_p_void m = li_cdata_cpp.malloc(256);
    li_cdata_cpp.memmove(m, s);
    byte[] ss = li_cdata_cpp.cdata(m, 7);
    string ss_string = utf8.GetString(ss);
    if (!ss_string.Equals("ABC\0abc"))
      throw new Exception("failed got: " + ss_string);
  }
}
